prover/crates/bin/prover_cli/tests/status/batch.rs

```diff
@@ -0,0 +1,1524 @@
+                BaseLayerCircuitType::VM,
+                2,
+            ),
+            (
+                ProverJobStatus::InProgress(ProverJobStatusInProgress::default()),
+                BaseLayerCircuitType::DecommitmentsFilter,
+                1,
+            ),
+        ]),
+        None,
+        None,
+        None,
+        None,
+        None,
+        batch_0,
+        &mut connection,
+    )
+    .await;
+
+    status_batch_0_expects(scenario_expected_stdout(
+        Status::Successful,
+        Some(Status::Successful),
+        Status::Successful,
+        Some(Status::InProgress),
+        Status::WaitingForProofs,
+        None,
+        Status::WaitingForProofs,
+        Status::WaitingForProofs,
+        Status::JobsNotFound,
+        batch_0,
+    ));
+
+    // Agg_round 1: all prover jobs successful. NWG queue.
+    create_scenario(
+        None,
+        None,
+        None,
+        Some(vec![(
+            ProverJobStatus::Successful(ProverJobStatusSuccessful::default()),
+            BaseLayerCircuitType::DecommitmentsFilter,
+            1,
+        )]),
+        Some(vec![
+            (WitnessJobStatus::Queued, BaseLayerCircuitType::VM),
+            (
+                WitnessJobStatus::Queued,
+                BaseLayerCircuitType::DecommitmentsFilter,
+            ),
+        ]),
+        None,
+        None,
+        None,
+        None,
+        batch_0,
+        &mut connection,
+    )
+    .await;
+
+    status_batch_0_expects(scenario_expected_stdout(
+        Status::Successful,
+        Some(Status::Successful),
+        Status::Successful,
+        Some(Status::Successful),
+        Status::Queued,
+        None,
+        Status::WaitingForProofs,
+        Status::WaitingForProofs,
+        Status::JobsNotFound,
+        batch_0,
+    ));
+
+    // NWG successful for VM circuit, agg_round 2 prover jobs created.
+    create_scenario(
+        None,
+        None,
+        None,
+        None,
+        Some(vec![
+            (
+                WitnessJobStatus::Successful(WitnessJobStatusSuccessful::default()),
+                BaseLayerCircuitType::VM,
+            ),
+            (
+                WitnessJobStatus::InProgress,
+                BaseLayerCircuitType::DecommitmentsFilter,
+            ),
+        ]),
+        Some(vec![(ProverJobStatus::Queued, BaseLayerCircuitType::VM, 1)]),
+        None,
+        None,
+        None,
+        batch_0,
+        &mut connection,
+    )
+    .await;
+
+    status_batch_0_expects(scenario_expected_stdout(
+        Status::Successful,
+        Some(Status::Successful),
+        Status::Successful,
+        Some(Status::Successful),
+        Status::InProgress,
+        Some(Status::Queued),
+        Status::WaitingForProofs,
+        Status::WaitingForProofs,
+        Status::JobsNotFound,
+        batch_0,
+    ));
+
+    // NWG successful, agg_round 2 prover jobs updated.
+    create_scenario(
```

Comment:
These function invocations are pretty hard to read.
Maybe make `Scenario` a structure and use a builder pattern there?